fiona/ogrext.pyx

```diff
@@ -525,16 +521,53 @@ cdef class Session:
         self.cogr_ds = NULL
 
     def get_fileencoding(self):
+        """DEPRECATED"""
         return self._fileencoding
 
-    def get_internalencoding(self):
-        if not self._encoding:
-            fileencoding = self.get_fileencoding()
-            self._encoding = (
-                OGR_L_TestCapability(
-                    self.cogr_layer, OLC_STRINGSASUTF8) and
-                'utf-8') or fileencoding
-        return self._encoding
+    def _get_fallback_encoding(self):
+        """Determine a format-specific fallback encoding to use when using OGR_F functions
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        str
+
+        """
+        if "Shapefile" in self.get_driver():
```

Comment:
Could this ever be "ESRI Shapefile"?

Reply:
For example:

```python
>>> import fiona
>>> src = fiona.open("tests/data/gre.shp", driver="ESRI Shapefile")
>>> src.session.get_driver()
'ESRI Shapefile'
```

Reply:
Yes, this was my attempt to shortcut and be future proof.

Reply:
Ah OK. Of course `"Shapefile" in` will match `"ESRI Shapefile"`.